lib/puppet/pops/types/type_calculator.rb

```diff
@@ -120,6 +120,30 @@ def self.infer(o)
     singleton.infer(o)
   end
 
+  # Infers a type if given object may have callable members, else returns nil.
+  # Caller must check for nil or if returned type supports members.
+  # This is a much cheaper call than doing a call to the general infer(o) method.
+  #
+  # @api private
+  def self.infer_callable_methods_t(o)
+    # If being a value that cannot have Pcore based methods callable from Puppet Language
+    if (o.is_a?(String) ||
+      o.is_a?(Numeric) ||
+      o.is_a?(TrueClass) ||
+      o.is_a?(FalseClass) ||
+      o.is_a?(Regexp) ||
+      o.instance_of?(Array) ||
+      o.instance_of?(Hash) ||
```

Comment:
As mentioned in https://github.com/puppetlabs/puppet/pull/7797#discussion_r340786433, `instance_of?` will match subclasses of `Array` and `Hash`, where the latter occurs in the acl module. It would be good to use `is_a?` consistently. That part will be handled in PUP-10105.